### ISSUE TITLE
🐛[BUG][Combat] TP can be lost on WS on a mob that moves out of range or is too far away.

### ISSUE THREAD

**Issue by JackWesleyNelson** [HorizonFFXI/HorizonXI-Issues#1520]
**What does this bug affect?**
When in combat, if a mob moves out of range, or is slightly out of range to begin with, it's possible to lose the TP when using a weapon skill (at least after using Sneak Attack, but I think in general).

**Steps to Reproduce**
Start out of range of a mob, and spam a weapon skill activation as you run towards them, or harder to reproduce without tooling, stand at the edge of a WS range and spam it as the mob begins to move out of range.

**Expected behavior**
The WS should either activate if it was in range within a small enough time period of the mob moving out of range, or the WS should not activate and the TP should not be consumed.

**Screenshots**
![image](https://user-images.githubusercontent.com/8739573/235314546-44a8c081-e72b-4bf0-a1d2-382a2d942f6b.png)

**Setup Information**
Launcher Version v1.14 
Occured as of the game version active on 4/29/2023 10:58 AM MDT

**Additional context**
Was mentioned as a comment to issue [386](https://github.com/HorizonFFXI/HorizonXI-Issues/issues/386) which has been marked as solved, but still occurs under the stated conditions for this issue.

**Comment by Shayemz** [comment]
I'm not a dev, but looking at the other report, I don't see a pull request linked to it that shows it was fixed. And if they did fix it, it looks like they might have fixed if the mob DIED then you didn't lose TP, as you can't WS on a dead mob. However, my memory of era is that your bug report is era accurate. You are supposed to lose TP if the mob moves out of range while you are readying. It was always frustrating in era, but led to you needing to be more conscious of when you chose to WS.

**Comment by JackWesleyNelson** [comment]
> I'm not a dev, but looking at the other report, I don't see a pull request linked to it that shows it was fixed. And if they did fix it, it looks like they might have fixed if the mob DIED then you didn't lose TP, as you can't WS on a dead mob. However, my memory of era is that your bug report is era accurate. You are supposed to lose TP if the mob moves out of range while you are readying. It was always frustrating in era, but led to you needing to be more conscious of when you chose to WS.

I didn't look for PR's, just saw it was closed as completed, without any other mention. Also, I agree that the fix likely could have been only for the mob dieing, since that was what they mentioned in the post, though the title is mentions "in some situations" which is more broad. I only linked the issue # in case someone thought that this behavior had been changed when that issue was closed. 

It certainly could have been in era behavior, but I have no way of testing that. I can say, I don't personally remember this ever occuring during my time playing retail, but that since starting on HorizonXI it's happened numerous times. 

If the behavior is era accurate, and the intended behavior on HorizonXI, then this will just be whenever someone makes that assesment. However, due to inconsistent server stability, I would argue this behavior is much more bug-like than skill-like and should be addressed in some way.
